app/index.js

```diff
@@ -20,6 +20,18 @@ app.config = config;
 app.plugins = plugins;
 app.getWindows = () => new Set([...windowSet]); // return a clone
 
+// function to retrive the last focused winow in windowSet;
+// added to app object in order to expose it to plugins.
+app.getLastFocusedWindow = () => {
+  let lastWindow = null;
```

Comment:
Implemented Array.prototype.reduce() method for getLastFocusedWindow, thank you again for the great advice 😄 I had only to handle empty windowSet case to avoid typeErrors.